rule.py

```diff
@@ -121,12 +121,16 @@ def to_be_or_not_to_be(self, old_pos, new_pos):
         pass 
 
     def is_interfered(self, old_pos, new_pos):
```

Comment:
Check when should this one return true and when should it returns false. True when it is interfered or when it is not interfered(allow to move). Make sure all the case follow that convention.